src-test/org/etools/j1939_84/controllers/part06/Part06Step03ControllerTest.java

```diff
@@ -120,10 +133,151 @@ public void testGetTotalSteps() {
     @Test
     public void testHappyPathNoFailures() {
 
+        DM12MILOnEmissionDTCPacket packet1 = DM12MILOnEmissionDTCPacket.create(0x01,
+                                                                               LampStatus.ON,
+                                                                               LampStatus.OFF,
+                                                                               LampStatus.OFF,
+                                                                               LampStatus.OFF,
+                                                                               DiagnosticTroubleCode.create(1569,
+                                                                                                            31,
+                                                                                                            0,
+                                                                                                            0));
+
+        dataRepository.putObdModule(new OBDModuleInformation(0x01));
+
+        when(diagnosticMessageModule.requestDM12(any(), eq(0x01))).thenReturn(new BusResult<>(false, packet1));
+
         runTest();
 
+        verify(diagnosticMessageModule).setJ1939(j1939);
+        verify(diagnosticMessageModule).requestDM12(any(), eq(0x01));
+
+        assertEquals("", listener.getResults());
         assertEquals("", listener.getMessages());
+        assertEquals("", listener.getMilestones());
+    }
+
+    @Test
+    public void testEmptyPacketFailure() {
+
+        dataRepository.putObdModule(new OBDModuleInformation(1));
+
+        when(diagnosticMessageModule.requestDM12(any(), eq(0x01)))
+                                                                  .thenReturn(new BusResult<>(false, Optional.empty()));
+
+        runTest();
+
+        verify(diagnosticMessageModule).requestDM12(any(), eq(0x01));
+
         assertEquals("", listener.getResults());
+        assertEquals("", listener.getMessages());
+
+        verify(mockListener).addOutcome(PART_NUMBER,
+                                        STEP_NUMBER,
+                                        FAIL,
+                                        "6.6.3.2.a - No ECU reported an active DTC and MIL on");
+        verify(mockListener).addOutcome(PART_NUMBER,
+                                        STEP_NUMBER,
+                                        FAIL,
+                                        "6.6.3.2.a - No ECU reported MIL on");
+        verify(mockListener).addOutcome(PART_NUMBER,
+                                        STEP_NUMBER,
+                                        FAIL,
+                                        "6.6.3.2.b - OBD module Engine #2 (1) did not provide a response to Global query and did not provide a NACK for the DS query");
+    }
+
+    @Test
+    public void testFailures() {
+        DM12MILOnEmissionDTCPacket packet1 = new DM12MILOnEmissionDTCPacket(Packet.create(PGN,
```

Comment:
going forward, we should be using packets where the inputs are understandable (e.g. `DM12MILOnEmissionDTCPacket.create()`)